modules/tracing/mule-tracer-internal-impl/src/main/java/org/mule/runtime/tracer/impl/context/NoExportOnEndInternalSpan.java

```diff
@@ -0,0 +1,133 @@
+
+import org.mule.runtime.api.profiling.tracing.Span;
+import org.mule.runtime.api.profiling.tracing.SpanDuration;
+import org.mule.runtime.api.profiling.tracing.SpanError;
+import org.mule.runtime.api.profiling.tracing.SpanIdentifier;
+import org.mule.runtime.tracer.api.span.ExportableInternalSpan;
+import org.mule.runtime.tracer.api.span.InternalSpan;
+import org.mule.runtime.tracer.api.span.error.InternalSpanError;
+import org.mule.runtime.tracer.api.span.exporter.SpanExporter;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+/**
+ * A {@link InternalSpan} wrapper to prevent exporting on end.
+ *
+ * @since 4.5.1, 4.6.0
+ */
+public class NoExportOnEndInternalSpan implements ExportableInternalSpan {
+
+  private final ExportableInternalSpan delegate;
+  private long endTime;
+
+  public NoExportOnEndInternalSpan(InternalSpan delegate) {
+    this.delegate = asExportable(delegate);
+  }
+
+  @Override
+  public ExportableInternalSpan updateChildSpanExporter(InternalSpan childInternalSpan) {
+    return this.delegate.updateChildSpanExporter(childInternalSpan);
+  }
+
+  @Override
+  public void updateRootName(String rootName) {
+    delegate.updateRootName(rootName);
+  }
+
+  @Override
+  public void setRootAttribute(String rootAttributeKey, String rootAttributeValue) {
+    delegate.setRootAttribute(rootAttributeKey, rootAttributeValue);
+  }
+
+  @Override
+  public void end() {
+    this.endTime = getDefault().now();
+  }
+
+  @Override
+  public void end(long endTime) {
+    this.delegate.end(endTime);
+  }
+
+  @Override
+  public void addError(InternalSpanError error) {
+    this.delegate.addError(error);
+  }
+
+  @Override
+  public void updateName(String name) {
+    this.delegate.updateName(name);
+  }
+
+  @Override
+  public SpanExporter getSpanExporter() {
+    return delegate.getSpanExporter();
+  }
+
+  @Override
+  public void forEachAttribute(BiConsumer<String, String> biConsumer) {
+    delegate.forEachAttribute(biConsumer);
+  }
+
+  @Override
+  public Map<String, String> serializeAsMap() {
+    return delegate.serializeAsMap();
+  }
+
+  @Override
+  public int getAttributesCount() {
+    return delegate.getAttributesCount();
+  }
+
+  @Override
+  public Span getParent() {
+    return delegate.getParent();
+  }
+
+  @Override
+  public SpanIdentifier getIdentifier() {
+    return delegate.getIdentifier();
+  }
+
+  @Override
+  public String getName() {
+    return delegate.getName();
+  }
+
+  @Override
+  public SpanDuration getDuration() {
+    return delegate.getDuration();
+  }
+
+  @Override
+  public List<SpanError> getErrors() {
+    return delegate.getErrors();
+  }
+
+  @Override
+  public boolean hasErrors() {
+    return delegate.hasErrors();
+  }
+
+  @Override
+  public void addAttribute(String key, String value) {
+    delegate.addAttribute(key, value);
+  }
+
+  @Override
+  public void export() {
+    this.delegate.end(endTime);
```

Comment:
We should call to export here, right?

Reply:
I've removed the ExportableInternalSpan interface because taking into account this request for change I found that it can work without adding this.